packages/react-scripts/config/paths.js

```diff
@@ -99,8 +101,10 @@ module.exports = {
   dotenv: resolveApp('.env'),
   appPath: resolveApp('.'),
   appBuild: resolveApp('build'),
-  appPublic: resolveApp('public'),
-  appHtml: resolveApp('public/index.html'),
+  appPublic: resolveApp(`public/${getServedPath(resolveApp('package.json'))}`),
+  appHtml: resolveOwn(
+    `public/${getServedPath(resolveApp('package.json'))}/index.html`
```

Comment:
tomorrow morning  I can adjust to this model, and get something in the way of tests. right now I can't even run `e2e` locally without a weird error.